spec/rails_cursor_pagination/cursor_spec.rb

```diff
@@ -156,6 +173,41 @@
       end
     end
 
+    context 'when decoding an encoded message with order_field :created_at' do
+      let(:record) { Post.create! id: 1, author: 'John', content: 'Post 1' }
+      let(:encoded) do
+        described_class.from_record(record: record, order_field: :created_at).encode
+      end
+
+      context 'and the order_field to decode is set to :id' do
+        subject(:decoded) do
+          described_class.decode(encoded_string: encoded)
+        end
+
+        it 'raises an InvalidCursorError' do
+          message = "The given cursor `#{encoded}` was decoded as " \
+                    "`[{\"seconds\": #{record.created_at.to_i},
+                    \"nanoseconds\": #{record.created_at.nsec}}, #{record.id}]` " \
+                    'but could not be parsed'
+          expect { decoded }.to raise_error(
+            ::RailsCursorPagination::InvalidCursorError,
+            message
+          )
+        end
+      end
+
+      context 'and the order_field to decode is set to :created_at' do
+        subject(:decoded) do
+          described_class.decode(encoded_string: encoded, order_field: :created_at)
+        end
+
+        it 'decodes the string succesfully' do
+          expect(decoded.id).to eq record.id
+          expect(decoded.order_field_value).to eq record.created_at
+        end
```

Comment:
I can't personally think of a way to do this, but it would be nice to have an assertion that tests nanosecond precision explicitly, instead of relying on an equality check on two `Time` objects, which inherently have that precision built in. Idk, I always hated testing timestamps so I think this is good.